src/main/java/io/vertx/core/eventbus/impl/EventBusImpl.java

```diff
@@ -227,11 +227,12 @@ protected MessageImpl createMessage(boolean send, String address, MultiMap heade
     return msg;
   }
 
-  protected <T> void addRegistration(String address, HandlerRegistration<T> registration,
+  protected <T> HandlerHolder<T> addRegistration(String address, HandlerRegistration<T> registration,
                                      boolean replyHandler, boolean localOnly) {
     Objects.requireNonNull(registration.getHandler(), "handler");
-    boolean newAddress = addLocalRegistration(address, registration, replyHandler, localOnly);
-    addRegistration(newAddress, address, replyHandler, localOnly, registration::setResult);
+    LocalRegistrationResult result = addLocalRegistration(address, registration, replyHandler, localOnly);
```

Comment:
Should be `LocalRegistrationResult<T> result = `

Reply:
yes

Reply:
but I think the main issue is that Handlers does not have a generic type for holders

Reply:
I will try to use generic if possible

Reply:
we might lose <T> because of Map<String, Handlers<?>>